tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/jsf/JSFModuleProjectNameValidationService.java

```diff
@@ -0,0 +1,140 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+package com.liferay.ide.project.core.modules.jsf;
+
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.project.core.ProjectCore;
+import com.liferay.ide.project.core.model.ProjectName;
+import com.liferay.ide.project.core.util.ValidationUtil;
+
+import java.io.File;
+
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.sapphire.FilteredListener;
+import org.eclipse.sapphire.PropertyContentEvent;
+import org.eclipse.sapphire.modeling.Path;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.platform.StatusBridge;
+import org.eclipse.sapphire.services.ValidationService;
+
+/**
+ * @author Simon Jiang
+ * @author Andy Wu
+ */
+public class JSFModuleProjectNameValidationService extends ValidationService
+{
+    private static final String PROJECT_NAME_REGEX = "[A-Za-z0-9_\\-.]+";
+
+    private FilteredListener<PropertyContentEvent> listener;
+
+    @Override
+    protected void initValidationService()
+    {
+        super.initValidationService();
+
+        listener = new FilteredListener<PropertyContentEvent>()
+        {
+            @Override
+            protected void handleTypedEvent( PropertyContentEvent event )
+            {
+                if( ! event.property().definition().equals( NewLiferayJSFModuleProjectOp.PROP_FINAL_PROJECT_NAME )
+                                && ! event.property().definition().equals( NewLiferayJSFModuleProjectOp.PROP_PROJECT_NAMES )
+                                && ! event.property().definition().equals( ProjectName.PROP_PROJECT_NAME ) )
+                {
+                    refresh();
+                }
+            }
+        };
+
+        op().attach( listener, "*" );
+    }
+
+    @Override
+    protected Status compute()
```

Comment:
This looks like it can be easily shared with other wizards.  Lets not just copy everything but try to share code where possible and follow DRY principle